lms/djangoapps/grades/signals.py

```diff
@@ -106,3 +110,50 @@ def submissions_score_reset_handler(sender, **kwargs):  # pylint: disable=unused
             u"Failed to process score_reset signal from Submissions API. "
             "user: %s, course_id: %s, usage_id: %s", user, course_id, usage_id
         )
+
+
+@receiver(SCORE_CHANGED)
+def recalculate_subsection_grade_handler(sender, **kwargs):  # pylint: disable=unused-argument
+    """
+    Consume the SCORE_CHANGED signal and trigger an update.
+    This method expects that the kwargs dictionary will contain the following
+    entries (See the definition of SCORE_CHANGED):
+       - points_possible: Maximum score available for the exercise
+       - points_earned: Score obtained by the user
+       - user_id: Integer User ID
+       - course_id: Unicode string representing the course
+       - usage_id: Unicode string indicating the courseware instance
+    """
+    if not settings.FEATURES.get('ENABLE_SUBSECTION_GRADES_SAVED', False):
+        return
+
+    points_possible = kwargs.get('points_possible', None)
+    points_earned = kwargs.get('points_earned', None)
+    course_id = kwargs.get('course_id', None)
+    usage_id = kwargs.get('usage_id', None)
+    user_id = kwargs.get('user_id', None)
+    if not all((user_id, course_id, usage_id)):
+        log.exception(
+            u"Failed to process SCORE_CHANGED signal. "
+            "points_possible: %s, points_earned: %s, user_id: %s, course_id: %s, "
+            "usage_id: %s", points_possible, points_earned, user_id, course_id, usage_id
+        )
+        return
+
+    course_key = CourseLocator.from_string(course_id)
+    usage_key = UsageKey.from_string(usage_id).replace(course_key=course_key)
+    student = user_by_anonymous_id(user_id)
+
+    if all((student, course_key, usage_key)):
+        from courseware.courses import get_course_by_id  # avoids circular import
+        course = get_course_by_id(course_key, depth=0)
+        course_structure_for_course = get_course_blocks(student, usage_key)
+        subsection = course_structure_for_course[usage_key]
+        from lms.djangoapps.grades.new.subsection_grade import SubsectionGradeFactory  # avoids circular import
```

Comment:
Rather than having multiple local imports here, consider changing the signature of the `SubsectionGrade.update` method and moving L148-152 into that file.